packages/@uppy/webcam/src/Webcam.tsx

```diff
@@ -1,73 +1,143 @@
 import { h } from 'preact'
 
 import { UIPlugin } from '@uppy/core'
+import type { Uppy, UIPluginOptions } from '@uppy/core'
+import type { DefinePluginOpts } from '@uppy/core/lib/BasePlugin.ts'
+import type { Body, Meta } from '@uppy/utils/lib/UppyFile.ts'
+import type { PluginTarget } from '@uppy/core/lib/UIPlugin.ts'
+import type { MinimalRequiredUppyFile } from '@uppy/core/lib/Uppy.ts'
 import getFileTypeExtension from '@uppy/utils/lib/getFileTypeExtension'
 import mimeTypes from '@uppy/utils/lib/mimeTypes'
 import isMobile from 'is-mobile'
 import canvasToBlob from '@uppy/utils/lib/canvasToBlob'
-import supportsMediaRecorder from './supportsMediaRecorder.js'
-import CameraIcon from './CameraIcon.jsx'
-import CameraScreen from './CameraScreen.jsx'
-import PermissionsScreen from './PermissionsScreen.jsx'
-
+import supportsMediaRecorder from './supportsMediaRecorder.ts'
+import CameraIcon from './CameraIcon.tsx'
+import CameraScreen from './CameraScreen.tsx'
+import PermissionsScreen from './PermissionsScreen.tsx'
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore We don't want TS to generate types for the package.json
 import packageJson from '../package.json'
-import locale from './locale.js'
+import locale from './locale.ts'
 
 /**
  * Normalize a MIME type or file extension into a MIME type.
  *
- * @param {string} fileType - MIME type or a file extension prefixed with `.`.
- * @returns {string|undefined} The MIME type or `undefined` if the fileType is an extension and is not known.
+ * @param fileType - MIME type or a file extension prefixed with `.`.
+ * @returns The MIME type or `undefined` if the fileType is an extension and is not known.
  */
-function toMimeType (fileType) {
+function toMimeType(fileType: string): string | undefined {
   if (fileType[0] === '.') {
-    return mimeTypes[fileType.slice(1)]
+    return mimeTypes[fileType.slice(1) as keyof typeof mimeTypes]
   }
   return fileType
 }
 
 /**
  * Is this MIME type a video?
- *
- * @param {string} mimeType - MIME type.
- * @returns {boolean}
  */
-function isVideoMimeType (mimeType) {
-  return /^video\/[^*]+$/.test(mimeType)
+function isVideoMimeType(mimeType?: string): boolean {
+  return /^video\/[^*]+$/.test(mimeType!)
 }
 
 /**
  * Is this MIME type an image?
- *
- * @param {string} mimeType - MIME type.
- * @returns {boolean}
  */
-function isImageMimeType (mimeType) {
-  return /^image\/[^*]+$/.test(mimeType)
+function isImageMimeType(mimeType?: string): boolean {
+  return /^image\/[^*]+$/.test(mimeType!)
```

Comment:
That kind of looks like asking for trouble? Either fix the types upstream or add `if (!mimeType) return false` or something

Reply:
I think a TypeScript conversion is almost always going to require a non-trivial amount of refactoring. It's really not just syntactic sugar on top of JS. Modifying the runtime as little as possible is going to cause you more work later and possible bugs in the meantime.